packages/mantine/src/components/copyToClipboard/CopyToClipboard.tsx

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+
+import {CheckSize24Px, CopySize24Px} from '@coveord/plasma-react-icons';
+import {TextInput, CopyButton, Tooltip, ActionIcon, createStyles, Textarea} from '@mantine/core';
+
+export enum CopyToClipboardVariant {
+    Button = 'button',
+    Text = 'text',
+    TextArea = 'textarea',
+}
+
+export interface CopyToClipboardProps {
+    /**
+     * The value to be copied to the clipboard.
+     */
+    value: string;
+    /**
+     * The variant type to use when displaying the component.
+     *
+     * @default "button"
+     */
+    variant?: 'button' | 'text' | 'textarea';
```

Comment:
Although I can see some value in having text input and textarea variants pre-made like this, I am not sure if it is really worth the maintenance of it. I can already see someone needing a missing prop on the Textinput or textarea.

Reply:
@gdostie what do you think of the compromise of only keeping a `withLabel` prop to display the value as well so that all of those occurrences can be consistent? If someone needs more props on their input or need to have it in a textArea they can do it themselves with the default "button" version of the component.

Reply:
Yup, lets go with that.